LongLookupTables/make_lookup_tables.py

```diff
@@ -0,0 +1,389 @@
+# -*- coding: utf-8 -*-
+
+"""
+Script to generate of the lookup tables problem.
+
+Running this script will save the following files in /dir/ or /dir/sample<i>/ if n_samples > 1:
+- train.tsv
+- validation.tsv
+- heldout_inputs.tsv
+- heldout_compositions
+- heldout_tables
+- new_compositions
+- longer_seen_1.tsv
+- longer_incremental_1.tsv
+- longer_new_1.tsv
+...
+- longer_seen_n.tsv
+- longer_incremental_n.tsv
+- longer_new_n.tsv
+
+with n is the max number of additional compositions in test compared to train.
+
+Help : `python make_lookup_tables.py -h`
+"""
+
+from __future__ import unicode_literals, division, absolute_import, print_function
+
+import sys
+import os
+import itertools
+import random
+import operator
+import warnings
+import argparse
+from functools import reduce
+
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+
+
+### MAIN ###
+def parse_arguments(args):
+    parser = argparse.ArgumentParser(description="Script to generate of the lookup tables problem.",
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument('-d', '--dir', default='.', help='Path to the directory where to save the generated data.')
+    parser.add_argument('-s', '--n-samples', type=int, default=1, help='Number of different samples to generate. If greater than 1, will save the files for a single sample in /dir/sample<i>/*.tsv.')
+
+    parser.add_argument('-v', '--validation-size', type=float, default=0.1, help='Percentage of training set to use as validation.')
+    parser.add_argument('-c', '--max-composition-train', type=int, default=4, help='Max length of compositions in training set.')
```

Comment:
perhaps use underscores everywhere instead of sometimes using '-'?

Reply:
Although I could do it, I would argue against. In CLI's it's actually standard practice to not use underscores in the parameters. For example `grep --line-number` or `zip --temp-path` or `vim --remote-wait`. By default `argparse` will store the variables with underscores in any case ! But normally it's considered more user friendly to use dashes :)

Reply:
hmm ok, it mostly occurred to me because I saw `not-intermediate` and `not_shuffle` in a row, which seemed a bit inconsistent to me.

Reply:
Ohh my bad it should all be with a dash :) good catch!